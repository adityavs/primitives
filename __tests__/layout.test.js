import React from 'react';
import renderer from 'react-test-renderer';

import { ConfigProvider, Column, Row, Float, H1 } from '../index'

describe('<Column>', () => {
  it('should render two H1 elements in column layout, centered', () => {
    const tree = renderer
      .create(
          <ConfigProvider>
            <Column center>
              <H1>Brian Kernighan</H1>
              <H1>Dennis Ritchie</H1>
            </Column>
          </ConfigProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})

describe('<Row>', () => {
  it('should render two H1 elements in row layout', () => {
    const tree = renderer
      .create(
          <ConfigProvider>
            <Row>
              <H1>Brian Kernighan</H1>
              <H1>Dennis Ritchie</H1>
            </Row>
          </ConfigProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})

describe('<Float>', () => {
  it('should render two H1 elements in float layout', () => {
    const tree = renderer
      .create(
          <ConfigProvider>
            <Float>
              <H1>Brian Kernighan</H1>
              <H1>Dennis Ritchie</H1>
            </Float>
          </ConfigProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})
