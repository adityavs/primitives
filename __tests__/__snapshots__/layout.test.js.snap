// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Column> should render two H1 elements in column layout, centered 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "flexDirection": "column",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "Helvetica",
          "fontWeight": "400",
        },
        Object {
          "fontSize": 32,
        },
        Object {},
      ]
    }
  >
    Brian Kernighan
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "Helvetica",
          "fontWeight": "400",
        },
        Object {
          "fontSize": 32,
        },
        Object {},
      ]
    }
  >
    Dennis Ritchie
  </Text>
</View>
`;

exports[`<Float> should render two H1 elements in float layout 1`] = `
<View
  style={
    Array [
      Object {
        "position": "absolute",
        "zIndex": 1,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "Helvetica",
          "fontWeight": "400",
        },
        Object {
          "fontSize": 32,
        },
        Object {},
      ]
    }
  >
    Brian Kernighan
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "Helvetica",
          "fontWeight": "400",
        },
        Object {
          "fontSize": 32,
        },
        Object {},
      ]
    }
  >
    Dennis Ritchie
  </Text>
</View>
`;

exports[`<Row> should render two H1 elements in row layout 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "Helvetica",
          "fontWeight": "400",
        },
        Object {
          "fontSize": 32,
        },
        Object {},
      ]
    }
  >
    Brian Kernighan
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "Helvetica",
          "fontWeight": "400",
        },
        Object {
          "fontSize": 32,
        },
        Object {},
      ]
    }
  >
    Dennis Ritchie
  </Text>
</View>
`;
